core/lib/libtomcrypt/src/math/fp/ltc_ecc_fp_mulmod.c

```diff
@@ -1436,8 +1436,10 @@ int ltc_ecc_fp_save_state(unsigned char **out, unsigned long *outlen)
     * and the mu INTEGER
     */
    cache_entry = XCALLOC(FP_ENTRIES*(2*(1U<<FP_LUT)+4)+3, sizeof(ltc_asn1_list));
-   if (cache_entry == NULL)
+   if (cache_entry == NULL){
+      LTC_MUTEX_UNLOCK(&ltc_ecc_fp_lock);
```

Comment:
I'd prefer something like:
``` C
err = CRYPT_MEM;
goto save_err;
```
Such a style makes it easier to match the locks and unlocks.

Reply:
Hi, it seems it can not write "goto save_err" in this case, which can lead to freeing a NULL pointer dangerously.
As you can see:
https://github.com/OP-TEE/optee_os/blob/203ee23d005b2cec2f21b5de334c5a246be32599/core/lib/libtomcrypt/src/math/fp/ltc_ecc_fp_mulmod.c#L1475-L1479

Reply:
Freeing `NULL` is harmless, but feel free to add another label just below the call to `XFREE()`.

Reply:
Indeed, freeing NULL is harmless according to [this.](https://stackoverflow.com/questions/6084218/is-it-good-practice-to-free-a-null-pointer-in-c/6084233).

Let me amend my commitment.